### DISCUSSION TITLE
Batch & Batch Continuations in Quartz.net similar to Hangfire approach

### DISCUSSION THREAD

**Discussion by HamedFathi** [quartznet/quartznet#1399] [Q&A] [answered]
I am looking for a way to simulate Batch & Batch Continuations in Quartz.net similar to the [Hangfire](https://www.hangfire.io/) approach.

```cs
// Batches
var batchId = BatchJob.StartNew(x =>
{
    x.Enqueue(() => Console.WriteLine("Job 1"));
    x.Enqueue(() => Console.WriteLine("Job 2"));
});

// Batch Continuations
BatchJob.ContinueBatchWith(batchId, x =>
{
    x.Enqueue(() => Console.WriteLine("Last Job"));
});
```

Is it possible to have something similar to these two HangFire features in Quartz.net?

Please share your knowledge and experience about batching via Quartz.net with me.

**Comment by lahma** [comment, marked as answer]
You could use [JobChainingJobListener](https://github.com/quartznet/quartznet/blob/35b8532dc005af2fc907b6d2516c2e4bf40f833b/src/Quartz/Listener/JobChainingJobListener.cs) to approach the problem, but if Hangfile suits your needs maybe you could use it from the get-go?

**Reply by HamedFathi** [reply]
Not really, I am trying to write a new library on top of your great quartz.net, please see [QuartzFire](https://github.com/HamedFathi/QuartzFire)

The last part is supporting Batch processes this is why I am asking.

I should check `JobChainingJobListener` thanks.

**Comment by lahma** [comment]
Cross-posted here: https://stackoverflow.com/questions/69881535/batch-batch-continuations-in-quartz-net-similar-to-hangfire-approach . Please make a note about doing so there won't be wasted efforts.

**Reply by HamedFathi** [reply]
I did not know the `Discussions` is active here, I deleted that post. Thanks.
